### PR TITLE
Implement GUI [PROTOTYPE]

### PR DESCRIPTION
**Pull request by PavlushaSource**

### CODE REVIEW

File: view/UI/window.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+func (d *CatsLayout) Layout(objects []fyne.CanvasObject, containerSize fyne.Size) {
+	scaleX := d.prevSize.Width / containerSize.Width
+	scaleY := d.prevSize.Height / containerSize.Height
+
+	for _, obj := range objects {
+
+		scaleVectorX := (d.ScaleCenter.X - obj.Position().X) * (scaleX - 1)
+		scaleVectorY := d.ScaleCenter.Y - obj.Position().Y*(scaleY-1)
```
Comment (tepa46) [inline view/UI/window.go]: (:)

File: view/UI/window.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		if r == '=' {
+			layout.Scale *= 1.1
+		} else if r == '-' {
+			layout.Scale /= 1.1
+		} else {
+			return
+		}
+		//fmt.Println("Size before resize", object.Size())
+		//currSize := object.Size()
+		nextSize := fyne.Size{Width: 1920 * layout.Scale, Height: 1080 * layout.Scale}
```
Comment (tepa46) [inline view/UI/window.go]: WindowH?

File: view/UI/bottom.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	"net/url"
+)
+
+func CreateBottom() fyne.CanvasObject {
+	texts := container.NewHBox(widget.NewLabel("tepa46, Arsene-Baitenov, PavlushaSoure"),
+		layout.NewSpacer(),
+		widget.NewButton("Source code", func() {
+			u, _ := url.Parse("https://github.com/PavlushaSource/Radar")
+			_ = fyne.CurrentApp().OpenURL(u)
+		}))
```
Comment (Arsene-Baitenov) [inline view/UI/bottom.go:18]:
```suggestion
		// TODO: Change to rectangle with link
		widget.NewButton("Source code", func() {
			u, _ := url.Parse("https://github.com/PavlushaSource/Radar")
			_ = fyne.CurrentApp().OpenURL(u)
		}))
```

File: view/UI/config.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
+package UI
+
+import (
+	"context"
+	"fmt"
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/dialog"
+	"fyne.io/fyne/v2/widget"
+	config2 "github.com/PavlushaSource/Radar/view/config"
```
Comment (Arsene-Baitenov) [inline view/UI/config.go]: nice lgtm (true (?))

File: view/UI/config.go
Comment (Arsene-Baitenov) [inline view/UI/config.go:1]: rename, e. g. mainMenu or mainMenuConfig

File: view/UI/config.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		selectDistanceType := widget.NewSelect([]string{"Euclidean", "Manhattan", "Curvilinear"}, func(s string) {
+			switch s {
+			case "Euclidean":
+				configBack.DistanceType = config2.Euclidean
+			case "Manhattan":
+				configBack.DistanceType = config2.Manhattan
+			case "Curvilinear":
+				configBack.DistanceType = config2.Curvilinear
+			}
+		})
```
Comment (Arsene-Baitenov) [inline view/UI/config.go]: string to constants

File: view/UI/config.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		selectDistanceType.PlaceHolder = ""
+		selectDistanceTypeDefaultValue := "Euclidean"
+		selectDistanceType.SetSelected(selectDistanceTypeDefaultValue)
+
+		form := &widget.Form{
+			Items: []*widget.FormItem{
+				{Text: "Count cats", Widget: catCount, HintText: "enter a positive number up to 5 * 10^4"},
```
Comment (Arsene-Baitenov) [inline view/UI/config.go]: mb move every variant to own constant entity

Comment (Arsene-Baitenov) [reply view/UI/config.go]:
```suggestion
				// TODO: mb move every variant to own constant entity
				{Text: "Count cats", Widget: catCount, HintText: "enter a positive number up to 5 * 10^4"},
```

File: view/UI/config.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+					w2.CenterOnScreen()
+					w2.Resize(configUI.WindowSize)
+
+					w2.Show()
+
+					configUI.InMainMenu = false
+				}
+			},
+			SubmitText: "Run",
+			CancelText: "Reset",
```
Comment (Arsene-Baitenov) [inline view/UI/config.go]: to constants

File: view/UI/window.go
Comment (Arsene-Baitenov) [inline view/UI/window.go:1]: rename, e. g. runWindow, runModelWindow

File: view/UI/window.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/PavlushaSource/Radar/view/config"
+	"github.com/PavlushaSource/Radar/view/domain"
+	"github.com/PavlushaSource/Radar/view/utils"
+	"time"
+)
+
+func CreateContentRunWindow(w1, w2 fyne.Window, backConfig *config.BackendConfig, UIConfig *config.UIConfig, ctx context.Context) fyne.CanvasObject {
+	//backEnd := GenerateBackendCats(50000)
+	backEnd := []domain.CatBackend{{X: 0, Y: 0, Color: domain.Red}, {X: 100, Y: 100, Color: domain.Purple}, {X: 1000, Y: 100, Color: domain.Blue}}
+	Cats := utils.CreateCats(backEnd, UIConfig.CatSize)
```
Comment (Arsene-Baitenov) [inline view/UI/window.go]: nice

File: view/UI/window.go [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+		//currentSize := obj.Size()
+		nextSize := fyne.NewSize(d.UIConfig.CatSize.Width*d.Scale, d.UIConfig.CatSize.Height*d.Scale)
+		obj.Resize(nextSize)
+		obj.Move(obj.Position().Add(moveCat))
+		//fmt.Println("Cat position", obj.Position())
+	}
+	d.prevSize = containerSize
+}
+
+func RegisterScaleRune(win fyne.Window, object fyne.CanvasObject, layout *CatsLayout) func(r rune) {
```
Comment (Arsene-Baitenov) [inline view/UI/window.go]: make scale not only by keyboard

Comment (Arsene-Baitenov) [reply view/UI/window.go]:
```suggestion
// TODO: make opportunity to call scale not only by keyboard
func RegisterScaleRune(win fyne.Window, object fyne.CanvasObject, layout *CatsLayout) func(r rune) {
```

File: view/UI/window.go [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+	return func(r rune) {
+		if r == '=' {
+			layout.Scale *= 1.1
+		} else if r == '-' {
+			layout.Scale /= 1.1
+		} else {
```
Comment (Arsene-Baitenov) [inline view/UI/window.go]: constants)))

File: view/UI/window.go [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+		//fmt.Println("Next size", nextSize.Width, nextSize.Height, "Scale", layout.Scale)
+
+		object.Resize(nextSize)
+		//fmt.Println("Size after resize", object.Size())
+		//object.Refresh()
+		//win.Canvas().Refresh(object)
+	}
+}
+
+func AnimateCat(A, B fyne.Position, cat fyne.CanvasObject, iterations float32) {
```
Comment (Arsene-Baitenov) [inline view/UI/window.go]: nice

File: view/UI/window.go [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+	vector := fyne.NewPos((A.X-B.X)/iterations, (A.Y-B.Y)/iterations)
```
Comment (Arsene-Baitenov) [inline view/UI/window.go]:
```suggestion
	// TODO: Add scale logic
	vector := fyne.NewPos((A.X-B.X)/iterations, (A.Y-B.Y)/iterations)
```

File: view/config/backendConfig.go
Comment (Arsene-Baitenov) [inline view/config/backendConfig.go:1]: rename, e. g. backendSettings, modelSettings

File: README.md [hunk 13]
```diff
@@ -1 +1,14 @@
...
+## GUI
+
+ - Fix start window size (smaller)
+ - Think about fixies main menu buttons positions on Macos
+ - Fix modeling window sizes
+ - Make modeling window resizable
+ - Reinit model on run (cat swim away)
+   - Stop modeling on exit to main menu??
+ - Set app icon
+ - Update app name (now Desctop)
```
Comment (Arsene-Baitenov) [inline README.md]:
```suggestion
 - Update app name (now Desctop)
 - Support two radiuses: For fighting and hissing
```

File: view/customTheme/dark.go [hunk 14]
```diff
@@ -0,0 +1,26 @@
+package customTheme
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"image/color"
+)
+
+type DarkTheme struct {
```
Comment (Arsene-Baitenov) [inline view/customTheme/dark.go:9]: nike

File: view/customTheme/light.go [hunk 15]
```diff
@@ -0,0 +1,26 @@
+package customTheme
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"image/color"
+)
+
+type LightTheme struct {
```
Comment (Arsene-Baitenov) [inline view/customTheme/light.go:9]: adidas

File: view/domain/domain.go [hunk 16]
```diff
@@ -0,0 +1,15 @@
+package domain
+
+type CatBackend struct {
+	X     float32
+	Y     float32
+	Color Color
+}
+
```
Comment (Arsene-Baitenov) [inline view/domain/domain.go:9]:
```suggestion
// TODO: Refactoring after chaining with backend model
```

**Review by tepa46 — APPROVED** [review]

**Review by Arsene-Baitenov — APPROVED** [review]

**Pull request merged**